frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/code_block/code_block_language_selector.dart

```diff
@@ -222,6 +222,9 @@ class _LanguageSelectionPopoverState extends State<_LanguageSelectionPopover> {
   void onArrowKey(AxisDirection direction) {
     if (filteredLanguages.isEmpty) return;
     final isUp = direction == AxisDirection.up;
+    if (selectedIndex < 0) {
```

Comment:
In which case, the selectedIndex will be less than 0? 

Alternatively, you can use clamp to ensure the selectedIndex is within the correct range.

```dart
    setState(() {
      if (isUp) { // 
        selectedIndex = selectedIndex == 0 ? length - 1 : selectedIndex - 1;
      } else {
        selectedIndex = selectedIndex == length - 1 ? 0 : selectedIndex + 1;
      }
     // <- clamp the selectedIndex here.
    });
```